install/helm/agones/templates/extensions.yaml

```diff
@@ -35,11 +47,11 @@ spec:
   service:
     name: agones-controller-service
     namespace: {{ .Release.Namespace }}
-        {{- if .Values.agones.controller.generateTLS }}
-  caBundle: {{ b64enc $ca.Cert }}
-        {{- else }}
+{{- if .Values.agones.controller.generateTLS }}
+  caBundle: {{ b64enc $certCert }}

```

Comment:
Before we were using the cert of the ca here (`ca.Cert`) and I think this is using the cert that was generated instead (`cert.Cert`). Shouldn't we be putting the cert from the certificate authority here and not the cert generated by the certificate authority?

Same for below on lines 81 and 126, whereas line 176 seems consistent with before.

Reply:
When not generating certs, there was used `server.crt`
```
caBundle: {{ default (.Files.Get "certs/server.crt") .Values.agones.controller.tlsCert | b64enc }}
```
which is not ca.cert, but cert.Cert equivalent. And this is when ambiguity happen.
`server.crt` and `server.key` are self signed cert, which can be, as far as I know, CA for itself. But the generated certs were using separated CA. I would prefer to maintain `Secret` clean, and not creating additional branches to it, so there is 2 possible ways I can follow:
- changing script for `server.key` and `server.crt` to also generate `ca.crt`
- generating self signed cert and use it as caBundle - `genSelfSignedCert` helm func
Is there any preference you would like me to implement?

Reply:
I used second approach